src/state.rs

```diff
@@ -210,14 +210,11 @@ impl DkgState {
     fn parts_into_acks<R: bls::rand::RngCore>(
         &mut self,
         parts: BTreeSet<IdPart>,
-        mut rng: R,
+        rng: &mut R,
```

Comment:
Why are we changing this back? Remember that `&mut R` is also `R`. So you can always pass `&mut rng` to this function.